crates/topos/src/components/tce/services/push_certificate.rs

```diff
@@ -203,3 +224,68 @@ impl Parser<NodeList> for InputFormat {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::check_certificate_delivery;
+    use rstest::*;
+    use std::time::Duration;
+    use topos_core::api::grpc::tce::v1::StatusRequest;
+    use topos_test_sdk::tce::create_network;
+    use tracing::{debug, info};
+
+    #[rstest]
+    #[test_log::test(tokio::test)]
+    #[timeout(Duration::from_secs(120))]
+    async fn assert_push_certificate_delivery() -> Result<(), Box<dyn std::error::Error>> {
+        let mut peers_context = create_network(5, vec![]).await;
+
+        let mut status: Vec<bool> = Vec::new();
+
+        for (_peer_id, client) in peers_context.iter_mut() {
+            let response = client
+                .console_grpc_client
+                .status(StatusRequest {})
+                .await
+                .expect("Can't get status");
+
+            status.push(response.into_inner().has_active_sample);
+        }
+
+        assert!(status.iter().all(|s| *s));
+
+        let nodes = peers_context
+            .iter()
+            .map(|peer| peer.1.api_entrypoint.clone())
+            .collect::<Vec<_>>();
+
+        debug!("Nodes used in test: {:?}", nodes);
+
+        let assertion = async move {
+            let peers: Vec<tonic::transport::Uri> = nodes
+                .into_iter()
+                .map(TryInto::try_into)
+                .collect::<Result<_, _>>()
+                .map_err(|e| format!("Unable to parse node list: {e}"))
+                .expect("Valid node list");
+
+            match check_certificate_delivery(5, peers, 30).await {
```

Comment:
Having huge timeout like that prevent us for detecting performance regression.

```suggestion
            match check_certificate_delivery(5, peers, 5).await {
```

Reply:
@Freyskeyd It times out always when I put 5 seconds